# Investment Case Group Project
### Spark Funds Investment Case Analysis

#### We will be using the numpy and pandas library for performing the analysis, thus importing them.

In [1]:
# Import the numpy and pandas packages

import numpy as np
import pandas as pd

## Checkpoint 1: Data Cleaning 1

### We are going to load the companies text file into a pandas dataframe, inspect it and clean it. 

In [2]:
# We will use Pandas read_csv method to read data from the provided text file. 
# We will use the separator as tabspace and encoding as latin_1 for this function.
# My code for importing the Company text file with proper encoding
companies = pd.read_csv("companies.txt",sep="\t",encoding="latin_1")

# Removing duplicate values if any
companies.drop_duplicates()

# Removing special characters by decoding them to ASCII values
companies.permalink=companies.permalink.str.encode('utf-8').str.decode('ascii', 'ignore').str.strip().str.upper()
companies.name=companies.name.str.encode('utf-8').str.decode('ascii', 'ignore')

In [3]:
# Getting rows & column count for companies dataframe
companies.shape

(66368, 10)

In [4]:
# Output of companies dataframe
companies

permalink  \
0                                    /ORGANIZATION/-FAME   
1                                 /ORGANIZATION/-QOUNTER   
2                    /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
3                                  /ORGANIZATION/0-6-COM   
4                         /ORGANIZATION/004-TECHNOLOGIES   
5                       /ORGANIZATION/01GAMES-TECHNOLOGY   
6                    /ORGANIZATION/0NDINE-BIOMEDICAL-INC   
7                                   /ORGANIZATION/0XDATA   
8                                        /ORGANIZATION/1   
9                              /ORGANIZATION/1-2-3-LISTO   
10                                 /ORGANIZATION/1-4-ALL   
11                        /ORGANIZATION/1-618-TECHNOLOGY   
12                           /ORGANIZATION/1-800-DENTIST   
13                           /ORGANIZATION/1-800-DOCTORS   
14              /ORGANIZATION/1-800-PUBLICRELATIONS-INC-   
15                            /ORGANIZATION/1-MAINSTREAM   
16                                 /ORGANIZATION/1-OF-99   
17                             /ORGANIZATION/10-20-MEDIA   
18                         /ORGANIZATION/10-MINUTES-WITH   
19                              /ORGANIZATION/1000-CORKS   
20                            /ORGANIZATION/1000-MARKETS   
21                                 /ORGANIZATION/1000CHI   
22                               /ORGANIZATION/1000LOOKZ   
23                            /ORGANIZATION/1000MEMORIES   
24                         /ORGANIZATION/1000MUSEUMS-COM   
25                              /ORGANIZATION/1001-MENUS   
26                      /ORGANIZATION/1001PHARMACIES-COM   
27                                 /ORGANIZATION/1006-TV   
28                                /ORGANIZATION/100DU-TV   
29                                /ORGANIZATION/100E-COM   
...                                                  ...   
66338                                /ORGANIZATION/ZYKIS   
66339                       /ORGANIZATION/ZYLIE-THE-BEAR   
66340                       /ORGANIZATION/ZYLUN-STAFFING   
66341                       /ORGANIZATION/ZYME-SOLUTIONS   
66342                             /ORGANIZATION/ZYMERGEN   
66343                              /ORGANIZATION/ZYMETIS   
66344                            /ORGANIZATION/ZYMEWORKS   
66345                             /ORGANIZATION/ZYMOCHEM   
66346                                /ORGANIZATION/ZYMPI   
66347                            /ORGANIZATION/ZYMR-INC-   
66348                                /ORGANIZATION/ZYNCD   
66349                               /ORGANIZATION/ZYNCRO   
66350              /ORGANIZATION/ZYNERBA-PHARMACEUTICALS   
66351            /ORGANIZATION/ZYNERGY-PROJECTS-SERVICES   
66352                                /ORGANIZATION/ZYNGA   
66353                             /ORGANIZATION/ZYNGENIA   
66354                              /ORGANIZATION/ZYNSTRA   
66355                           /ORGANIZATION/ZYOMYX-INC   
66356                                 /ORGANIZATION/ZYPE   
66357                               /ORGANIZATION/ZYPSEE   
66358                     /ORGANIZATION/ZYRAZ-TECHNOLOGY   
66359                                /ORGANIZATION/ZYRRA   
66360                               /ORGANIZATION/ZYSTOR   
66361                           /ORGANIZATION/ZYTOPROTEC   
66362                                /ORGANIZATION/ZZISH   
66363  /ORGANIZATION/ZZNODE-SCIENCE-AND-TECHNOLOGY-CO...   
66364                          /ORGANIZATION/ZZZZAPP-COM   
66365                                 /ORGANIZATION/ERON   
66366                               /ORGANIZATION/ASYS-2   
66367  /ORGANIZATION/NOVATIFF-REKLAM-VE-TANTM-HIZMETL...   

                                          name  \
0                                        #fame   
1                                     :Qounter   
2                       (THE) ONE of THEM,Inc.   
3                                      0-6.com   
4                             004 Technologies   
5                         

### Now, we are going to load the rounds2 csv file into a pandas dataframe, inspect it and clean it. 

In [5]:
# We will use Pandas read_csv method to read data from the provided csv file. 
# We will use the default separator and encoding as latin_1 for this function.
# My code for importing the Rounds2 csv file with proper encoding
rounds2 = pd.read_csv("rounds2.csv",encoding="latin_1")

# Removing duplicate values if any
rounds2.drop_duplicates()

# Removing special characters by decoding them to ASCII values
rounds2.company_permalink=rounds2.company_permalink.str.encode('utf-8').str.decode('ascii', 'ignore').str.strip().str.upper()

In [6]:
# Getting rows & column count for rounds2 dataframe
rounds2.shape

(114949, 6)

In [7]:
# Output of rounds2 dataframe
rounds2

company_permalink  \
0                                     /ORGANIZATION/-FAME   
1                                  /ORGANIZATION/-QOUNTER   
2                                  /ORGANIZATION/-QOUNTER   
3                     /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                                   /ORGANIZATION/0-6-COM   
5                          /ORGANIZATION/004-TECHNOLOGIES   
6                        /ORGANIZATION/01GAMES-TECHNOLOGY   
7                     /ORGANIZATION/0NDINE-BIOMEDICAL-INC   
8                     /ORGANIZATION/0NDINE-BIOMEDICAL-INC   
9                                    /ORGANIZATION/0XDATA   
10                                   /ORGANIZATION/0XDATA   
11                                   /ORGANIZATION/0XDATA   
12                                   /ORGANIZATION/0XDATA   
13                                        /ORGANIZATION/1   
14                                        /ORGANIZATION/1   
15                                        /ORGANIZATION/1   
16                              /ORGANIZATION/1-2-3-LISTO   
17                                  /ORGANIZATION/1-4-ALL   
18                         /ORGANIZATION/1-618-TECHNOLOGY   
19                            /ORGANIZATION/1-800-DENTIST   
20                            /ORGANIZATION/1-800-DOCTORS   
21               /ORGANIZATION/1-800-PUBLICRELATIONS-INC-   
22                             /ORGANIZATION/1-MAINSTREAM   
23                                  /ORGANIZATION/1-OF-99   
24                              /ORGANIZATION/10-20-MEDIA   
25                              /ORGANIZATION/10-20-MEDIA   
26                              /ORGANIZATION/10-20-MEDIA   
27                              /ORGANIZATION/10-20-MEDIA   
28                          /ORGANIZATION/10-MINUTES-WITH   
29                          /ORGANIZATION/10-MINUTES-WITH   
...                                                   ...   
114919                              /ORGANIZATION/ZYNSTRA   
114920                           /ORGANIZATION/ZYOMYX-INC   
114921                           /ORGANIZATION/ZYOMYX-INC   
114922                           /ORGANIZATION/ZYOMYX-INC   
114923                           /ORGANIZATION/ZYOMYX-INC   
114924                                 /ORGANIZATION/ZYPE   
114925                                 /ORGANIZATION/ZYPE   
114926                               /ORGANIZATION/ZYPSEE   
114927                               /ORGANIZATION/ZYPSEE   
114928                     /ORGANIZATION/ZYRAZ-TECHNOLOGY   
114929                     /ORGANIZATION/ZYRAZ-TECHNOLOGY   
114930                     /ORGANIZATION/ZYRAZ-TECHNOLOGY   
114931                     /ORGANIZATION/ZYRAZ-TECHNOLOGY   
114932                                /ORGANIZATION/ZYRRA   
114933                                /ORGANIZATION/ZYRRA   
114934                                /ORGANIZATION/ZYRRA   
114935                                /ORGANIZATION/ZYRRA   
114936                               /ORGANIZATION/ZYSTOR   
114937                           /ORGANIZATION/ZYTOPROTEC   
114938                                /ORGANIZATION/ZZISH   
114939                                /ORGANIZATION/ZZISH   
114940                                /ORGANIZATION/ZZISH   
114941  /ORGANIZATION/ZZNODE-SCIENCE-AND-TECHNOLOGY-CO...   
114942                          /ORGANIZATION/ZZZZAPP-COM   
114943                          /ORGANIZATION/ZZZZAPP-COM   
114944                          /ORGANIZATION/ZZZZAPP-COM   
114945                          /ORGANIZATION/ZZZZAPP-COM   
114946                                 /ORGANIZATION/ERON   
114947                               /ORGANIZATION/ASYS-2   
114948  /ORGANIZATION/NOVATIFF-REKLAM-VE-TANTM-HIZMETL...   

                                funding_round_permalink   funding_round_type  \
0       /funding-round/9a01d05418af9f794eebff7ace91f638              venture   
1       /funding-round/22dacff496eb7acb2b901dec1dfe5633              venture   
2       /funding-

### Preparing Table 1.1 using the two dataframes i.e. *companies* and *rounds2*

In [8]:
#Results Expected: Table 1.1

#Making set of unique values from rounds2.company_permalink & companies.permalink
u_c_in_r = set(rounds2.company_permalink)
u_c_in_c = set(companies.permalink)
# Counting the values
print("unique companies present in rounds2: ",len(u_c_in_r))   #1. 66368
print("unique companies present in companies: ",len(u_c_in_c)) #2. 66368

#3. permalink

# Checking if all companies in rounds2 are present in companies dataframe
print("Are all the companies in rounds2 a subset of all companies in companies dataframe: ",set(u_c_in_r).issubset(u_c_in_c)) # True 
print("Count of companies in the rounds2 file which are not present in companies: ",len(u_c_in_r.difference(u_c_in_c))) # 0

#4. N

# Merging the two dataframes
master_frame=pd.merge(rounds2, companies, left_on='company_permalink', right_on='permalink')

print("number of rows & columns in master_frame before data cleaning", master_frame.shape)

# Frher cleaning data in the merged dataframe
master_frame["raised_amount_usd"].describe()
# Upon analysis of "raised_amount_usd" field, we came to know that there are multiple entries which contain 0 which isn't correct.
# Thus we ignore the rows with 0 as "raised_amount_usd" for the below calculations.
master_frame = master_frame[master_frame['raised_amount_usd'] != 0]
# Dropping rows where 'funding_round_type','raised_amount_usd','category_list','country_code' are NaN, since these fields are necessary.
master_frame = master_frame.dropna(subset=['funding_round_type','raised_amount_usd','category_list','country_code'])

#Getting rows & column count for master_frame dataframe
print("number of rows & columns in master_frame after data cleaning", master_frame.shape) #5. 88166

unique companies present in rounds2:  66368
unique companies present in companies:  66368
Are all the companies in rounds2 a subset of all companies in companies dataframe:  True
Count of companies in the rounds2 file which are not present in companies:  0
number of rows & columns in master_frame before data cleaning (114949, 16)
number of rows & columns in master_frame after data cleaning (88166, 16)


### Table 1.1: Understand the Data Set 
| Question | Response |
| -------- | -------- |
| How many unique companies are present in rounds2? | 66368 |
| How many unique companies are present in companies? | 66368 |
| In the companies data frame, which column can be used as the unique key for each company? Write the name of the column. | permalink |
| Are there any companies in the rounds2 file which are not present in companies? Answer yes or no: Y/N | N |
| Merge the two data frames so that all variables (columns) in the companies frame are added to the rounds2 data frame.Name the merged frame master_frame. How many observations are present in master_frame? | 88166 |
**Note:** 88166 is the number of observations present after data cleaning, initially we had 114949 observations in master_frame

## Checkpoint 2: Funding Type Analysis

### Analysing which investment type is the most suitable for Spark Funds.

In [9]:
# Calculate the average investment amount for each of the four funding types (venture, angel, seed, and private equity) and report the answers in Table 2.1

print("Average investment amount for each of the four funding types:")
print("venture: {:,.2f}".format(round(master_frame[(master_frame['funding_round_type'] == "venture")]["raised_amount_usd"].mean(),2)))               #1. 11,729,865.71
print("angel: {:,.2f}".format(round(master_frame[(master_frame['funding_round_type'] == "angel")]["raised_amount_usd"].mean(),2)))                   #2. 977,572.63
print("seed: {:,.2f}".format(round(master_frame[(master_frame['funding_round_type'] == "seed")]["raised_amount_usd"].mean(),2)))                     #3. 750,319.05
print("private_equity: {:,.2f}".format(round(master_frame[(master_frame['funding_round_type'] == "private_equity")]["raised_amount_usd"].mean(),2))) #4. 73,979,134.15

Average investment amount for each of the four funding types:
venture: 11,729,865.71
angel: 977,572.63
seed: 750,319.05
private_equity: 73,979,134.15


### Table 2.1: Average Values of Investments for Each of these Funding Types 

| Question | Response |
| -------- | -------- |
| Average funding amount of venture type | 11729865.71 |
| Average funding amount of angel type | 977572.63 |
| Average funding amount of seed type | 750319.05 |
| Average funding amount of private equity type | 73979134.15 |
| Considering that Spark Funds wants to invest between 5 to 15 million USDper investment round, which investment type is the most suitable for it? | venture |

## Checkpoint 3: Country Analysis

### Analysing which countries are most suitable for Spark Funds.

In [10]:
# Finding country list according to the requirement:
# top nine countries which have received the highest total funding (across ALL sectors for the chosen investment type)
condition = master_frame[(master_frame['funding_round_type'] == "venture")].groupby(by=['country_code']).raised_amount_usd.sum().sort_values(ascending = False)[0:9]
country_list = list(condition.index)

# Making top9 DataFrame using marster_frame filtered upon venture and country_list
top9 = master_frame[(master_frame['funding_round_type'] == "venture")&(master_frame['country_code'].isin(country_list))]

# Verifying if we got dataframe only, as specifically mentioned in the question.
print(type(top9))

# Displaying country-wise sum of raised_amount_usd in descending order
# We have used the lambda function to convert scientific notation to more readable format
top9.groupby('country_code')['raised_amount_usd'].sum().sort_values(ascending=False).map(lambda x: '{:,.2f}'.format(x))

<class 'pandas.core.frame.DataFrame'>


country_code
USA    420,068,029,342.00
CHN     39,338,918,773.00
GBR     20,072,813,004.00
IND     14,261,508,718.00
CAN      9,482,217,668.00
FRA      7,226,851,352.00
ISR      6,854,350,477.00
DEU      6,306,921,981.00
JPN      3,167,647,127.00
Name: raised_amount_usd, dtype: object

### Table 3.1: Analysing the Top 3 English-Speaking Countries

| Question | Response |
| -------- | -------- |
| 1. Top English-speaking country | USA |
| 2. Second English-speaking country | GBR |
| 3. Third English-speaking country | IND |
**Note:** We used [this](http://www.emmir.org/fileadmin/user_upload/admission/Countries_where_English_is_an_official_language.pdf) link as suggested in requirements to filter out English-speaking country

## Checkpoint 4: Sector Analysis 1

### Preparing for analysis of which sectors are most suitable for Spark Funds.

#### Adding *primary_sector* information *master_frame* dataframe according to requirement

In [11]:
# Extracting the primary sector of each category list from the category_list column
master_frame['primary_sector'] = master_frame['category_list'].apply(lambda x: str(x).split('|')[0])

# Output of updated master_frame dataframe
master_frame.head()

company_permalink  \
0                  /ORGANIZATION/-FAME   
2               /ORGANIZATION/-QOUNTER   
4                /ORGANIZATION/0-6-COM   
6     /ORGANIZATION/01GAMES-TECHNOLOGY   
7  /ORGANIZATION/0NDINE-BIOMEDICAL-INC   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
6  /funding-round/7d53696f2b4f607a2f2a8cbb83d01839        undisclosed   
7  /funding-round/2b9d3ac293d5cdccbecff5c8cb0f327d               seed   

  funding_round_code   funded_at  raised_amount_usd  \
0                  B  05-01-2015         10000000.0   
2                NaN  01-03-2014           700000.0   
4                  A  19-03-2008          2000000.0   
6                NaN  01-07-2014            41250.0   
7                NaN  11-09-2009            43360.0   

                             permalink                    name  \
0                  /ORGANIZATION/-FAME                   #fame   
2               /ORGANIZATION/-QOUNTER                :Qounter   
4                /ORGANIZATION/0-6-COM                 0-6.com   
6     /ORGANIZATION/01GAMES-TECHNOLOGY      01Games Technology   
7  /ORGANIZATION/0NDINE-BIOMEDICAL-INC  Ondine Biomedical Inc.   

             homepage_url                                      category_list  \
0      http://livfame.com                                              Media   
2  http://www.qounter.com  Application Platforms|Real Time|Social Network...   
4      http://www.0-6.com                                        Curated Web   
6  http://www.01games.hk/                                              Games   
7    http://ondinebio.com                                      Biotechnology   

      status country_code state_code      region           city  founded_at  \
0  operating          IND         16      Mumbai         Mumbai         NaN   
2  operating          USA         DE  DE - Other  Delaware City  04-09-2014   
4  operating          CHN         22     Beijing        Beijing  01-01-2007   
6  operating          HKG        NaN   Hong Kong      Hong Kong         NaN   
7  operating          CAN         BC   Vancouver      Vancouver  01-01-1997   

          primary_sector  
0                  Media  
2  Application Platforms  
4            Curated Web  
6                  Games  
7          Biotechnology

#### Loading and cleaning the *mapping table* and storing it in *mapping* dataframe

In [12]:
mapping = pd.read_csv("mapping.csv",encoding="latin_1") # My code for importing the mapping csv file
# Reviewing data in mapping dataframe
mapping

# Data cleaning
# Upon close observation, we find multiple '0' in middle of text making the text meaningless in column category_list
# However, upon closer observation, we found that replacing '0' with 'na' makes the text meaningful in each case in column category_list
# Thus replacing each occurence of '0' which is followed by text with 'na' in column category_list
# mapping function to provide the value for column main_sector based upon the values in mapping file
def clean_data(val):
    if ('0' in str(val[0])) and ('.0' not in str(val[0])):
        return str(val[0]).replace('0', 'na')
    else:
        return val[0]

# Using clean_data to clean the category_list column values based upon our observation
mapping['category_list'] = mapping[['category_list']].apply(lambda val:clean_data(val), axis=1)

# Output of mapping dataframe
mapping.head()

category_list  Automotive & Sports  Blanks  Cleantech / Semiconductors  \
0            NaN                    0       1                           0   
1             3D                    0       0                           0   
2    3D Printing                    0       0                           0   
3  3D Technology                    0       0                           0   
4     Accounting                    0       0                           0   

   Entertainment  Health  Manufacturing  News, Search and Messaging  Others  \
0              0       0              0                           0       0   
1              0       0              1                           0       0   
2              0       0              1                           0       0   
3              0       0              1                           0       0   
4              0       0              0                           0       0   

   Social, Finance, Analytics, Advertising  
0                                        0  
1                                        0  
2                                        0  
3                                        0  
4                                        1

#### Mapping each company with one of the main sectors using the *master_frame* and *mapping* dataframes

In [13]:
# Converting the mapping keys (mapping.category_list and companies.primary_sector) to same case (upper case)
mapping['category_list'] = mapping.apply(lambda val:str(val['category_list']).strip().upper(), axis=1)
master_frame['primary_sector'] = master_frame.apply(lambda val:str(val['primary_sector']).strip().upper(), axis=1)

# Function to return the main_sector based upon row values
def get_main_sector (row):
    if row['Automotive & Sports'] == 1 : return 'Automotive & Sports'
    elif row['Blanks'] == 1 : return 'Blanks'
    elif row['Cleantech / Semiconductors'] == 1 : return 'Cleantech / Semiconductors'
    elif row['Entertainment'] == 1: return 'Entertainment'
    elif row['Health']  == 1: return 'Health'
    elif row['Manufacturing'] == 1: return 'Manufacturing'
    elif row['News, Search and Messaging'] == 1: return 'News, Search and Messaging'
    elif row['Others'] == 1: return 'Others'
    elif row['Social, Finance, Analytics, Advertising'] == 1: return 'Social, Finance, Analytics, Advertising'
    if row['Automotive & Sports']+row['Cleantech / Semiconductors']+row['Entertainment']+row['Health']+row['Manufacturing']+row['News, Search and Messaging']+row['Others']+row['Social, Finance, Analytics, Advertising'] > 1: return 'Multiple'
    if row['Automotive & Sports']+row['Cleantech / Semiconductors']+row['Entertainment']+row['Health']+row['Manufacturing']+row['News, Search and Messaging']+row['Others']+row['Social, Finance, Analytics, Advertising'] < 1: return 'Blanks'

# Using get_main_sector to populate main_sector column of mapping datafield
mapping['main_sector']=mapping.apply(lambda row: get_main_sector(row),axis=1)

# Merging the companies and mapping dataframe to get companies along with their main sector, calling this dataframe company_sector
master_frame=pd.merge(master_frame, mapping[['category_list','main_sector']], left_on='primary_sector', right_on='category_list', how='left')

master_frame.head()

company_permalink  \
0                  /ORGANIZATION/-FAME   
1               /ORGANIZATION/-QOUNTER   
2                /ORGANIZATION/0-6-COM   
3     /ORGANIZATION/01GAMES-TECHNOLOGY   
4  /ORGANIZATION/0NDINE-BIOMEDICAL-INC   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
2  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
3  /funding-round/7d53696f2b4f607a2f2a8cbb83d01839        undisclosed   
4  /funding-round/2b9d3ac293d5cdccbecff5c8cb0f327d               seed   

  funding_round_code   funded_at  raised_amount_usd  \
0                  B  05-01-2015         10000000.0   
1                NaN  01-03-2014           700000.0   
2                  A  19-03-2008          2000000.0   
3                NaN  01-07-2014            41250.0   
4                NaN  11-09-2009            43360.0   

                             permalink                    name  \
0                  /ORGANIZATION/-FAME                   #fame   
1               /ORGANIZATION/-QOUNTER                :Qounter   
2                /ORGANIZATION/0-6-COM                 0-6.com   
3     /ORGANIZATION/01GAMES-TECHNOLOGY      01Games Technology   
4  /ORGANIZATION/0NDINE-BIOMEDICAL-INC  Ondine Biomedical Inc.   

             homepage_url                                    category_list_x  \
0      http://livfame.com                                              Media   
1  http://www.qounter.com  Application Platforms|Real Time|Social Network...   
2      http://www.0-6.com                                        Curated Web   
3  http://www.01games.hk/                                              Games   
4    http://ondinebio.com                                      Biotechnology   

      status country_code state_code      region           city  founded_at  \
0  operating          IND         16      Mumbai         Mumbai         NaN   
1  operating          USA         DE  DE - Other  Delaware City  04-09-2014   
2  operating          CHN         22     Beijing        Beijing  01-01-2007   
3  operating          HKG        NaN   Hong Kong      Hong Kong         NaN   
4  operating          CAN         BC   Vancouver      Vancouver  01-01-1997   

          primary_sector        category_list_y                 main_sector  
0                  MEDIA                  MEDIA               Entertainment  
1  APPLICATION PLATFORMS  APPLICATION PLATFORMS  News, Search and Messaging  
2            CURATED WEB            CURATED WEB  News, Search and Messaging  
3                  GAMES                  GAMES               Entertainment  
4          BIOTECHNOLOGY          BIOTECHNOLOGY  Cleantech / Semiconductors

## Checkpoint 5: Sector Analysis 2

### Analysing which sectors are most suitable for Spark Funds considering country and amount of investment.

#### Creating dataframes D1, D2 and D3 as per given requirements.

In [14]:
# Dataframe for contry_code "USA", funding_round_type "venture" and raised_amount_usd in inclusive range of 5 million to 15 million
D1 = master_frame[(master_frame['country_code']=="USA") & (master_frame['funding_round_type']=="venture") & (master_frame['raised_amount_usd']>=5000000) & (master_frame['raised_amount_usd']<=15000000) ]

# Dataframe for contry_code "GBR", funding_round_type "venture" and raised_amount_usd in inclusive range of 5 million to 15 million
D2 = master_frame[(master_frame['country_code']=="GBR") & (master_frame['funding_round_type']=="venture") & (master_frame['raised_amount_usd']>=5000000) & (master_frame['raised_amount_usd']<=15000000) ]

# Dataframe for contry_code "IND", funding_round_type "venture" and raised_amount_usd in inclusive range of 5 million to 15 million
D3 = master_frame[(master_frame['country_code']=="IND") & (master_frame['funding_round_type']=="venture") & (master_frame['raised_amount_usd']>=5000000) & (master_frame['raised_amount_usd']<=15000000) ]

#### Analysing total number (or count) of investments and total amount invested in each main sector for USA

In [15]:
print("Total number of investments (count): ", D1['funding_round_permalink'].count()) # 12064
print("Total amount of investment (USD): {:,.2f}".format(D1['raised_amount_usd'].sum())) # 107,766,584,223.00
print(D1.groupby(['main_sector'])['funding_round_permalink'].count().sort_values(ascending=False)[0:3])
# Others                                     2950
# Social, Finance, Analytics, Advertising    2714
# Cleantech / Semiconductors                 2350
print("For the top sector count-wise, company that received the highest investment: ", 
      D1[(D1['main_sector']=="Others")].groupby(['name'])['raised_amount_usd'].sum().sort_values(ascending=False).index[0])                                  # Virtustream
print("For the second-best sector count-wise, company that received the highest investment: ",
      D1[(D1['main_sector']=="Social, Finance, Analytics, Advertising")].groupby(['name'])['raised_amount_usd'].sum().sort_values(ascending=False).index[0]) # SST Inc. (Formerly ShotSpotter)

Total number of investments (count):  12064
Total amount of investment (USD): 107,766,584,223.00
main_sector
Others                                     2950
Social, Finance, Analytics, Advertising    2714
Cleantech / Semiconductors                 2350
Name: funding_round_permalink, dtype: int64
For the top sector count-wise, company that received the highest investment:  Virtustream
For the second-best sector count-wise, company that received the highest investment:  SST Inc. (Formerly ShotSpotter)


#### Analysing total number (or count) of investments and total amount invested in each main sector for GBR

In [16]:
print("Total number of investments (count): ", D2['funding_round_permalink'].count()) # 621
print("Total amount of investment (USD): {:,.2f}".format(D2['raised_amount_usd'].sum())) # 5,379,078,691.00
print(D2.groupby(['main_sector'])['funding_round_permalink'].count().sort_values(ascending=False)[0:3])
# Others                                     147
# Social, Finance, Analytics, Advertising    133
# Cleantech / Semiconductors                 130
print("For the top sector count-wise, company that received the highest investment: ", 
      D2[(D2['main_sector']=="Others")].groupby(['name'])['raised_amount_usd'].sum().sort_values(ascending=False).index[0])                                  # Electric Cloud
print("For the second-best sector count-wise, company that received the highest investment: ",
      D2[(D2['main_sector']=="Social, Finance, Analytics, Advertising")].groupby(['name'])['raised_amount_usd'].sum().sort_values(ascending=False).index[0]) # Celltick Technologies

Total number of investments (count):  621
Total amount of investment (USD): 5,379,078,691.00
main_sector
Others                                     147
Social, Finance, Analytics, Advertising    133
Cleantech / Semiconductors                 130
Name: funding_round_permalink, dtype: int64
For the top sector count-wise, company that received the highest investment:  Electric Cloud
For the second-best sector count-wise, company that received the highest investment:  Celltick Technologies


#### Analysing total number (or count) of investments and total amount invested in each main sector for IND

In [17]:
print("Total number of investments (count): ", D3['funding_round_permalink'].count()) # 328
print("Total amount of investment (USD): {:,.2f}".format(D3['raised_amount_usd'].sum())) # 2,949,543,602.00
print(D3.groupby(['main_sector'])['funding_round_permalink'].count().sort_values(ascending=False)[0:3])
# Others                                     110
# Social, Finance, Analytics, Advertising     60
# News, Search and Messaging                  52
print("For the top sector count-wise, company that received the highest investment: ", 
      D3[(D3['main_sector']=="Others")].groupby(['name'])['raised_amount_usd'].sum().sort_values(ascending=False).index[0])                                  # FirstCry.com
print("For the second-best sector count-wise, company that received the highest investment: ",
      D3[(D3['main_sector']=="Social, Finance, Analytics, Advertising")].groupby(['name'])['raised_amount_usd'].sum().sort_values(ascending=False).index[0]) # Manthan Systems

Total number of investments (count):  328
Total amount of investment (USD): 2,949,543,602.00
main_sector
Others                                     110
Social, Finance, Analytics, Advertising     60
News, Search and Messaging                  52
Name: funding_round_permalink, dtype: int64
For the top sector count-wise, company that received the highest investment:  FirstCry.com
For the second-best sector count-wise, company that received the highest investment:  Manthan Systems


### Table 5.1 : Sector-wise Investment Analysis

| Question | USA | GBR | IND |
| -------- | --- | --- | --- |
| 1. Total number of investments (count) | 12064 | 621 | 328 |
| 2. Total amount of investment (USD) | 1,07,76,65,84,223 | 5,37,90,78,691 | 2,94,95,43,602 |
| 3. Top sector (based on count of investments)	| Others | Others | Others |
| 4. Second-best sector (based on count of investments) | Social, Finance, Analytics, Advertising | Social, Finance, Analytics, Advertising | Social, Finance, Analytics, Advertising |
| 5. Third-best sector (based on count of investments) | Cleantech / Semiconductors	| Cleantech / Semiconductors | News, Search and Messaging |
| 6. Number of investments in the top sector (refer to point 3) | 2950 | 147 | 110 |
| 7. Number of investments in the second-best sector (refer to point 4) | 2714 | 133 | 60 |
| 8. Number of investments in the third-best sector (refer to point 5) | 2350 | 130 | 52 |
| 9. For the top sector count-wise (point 3), which company received the highest investment? | Virtustream | Electric Cloud | FirstCry.com |
| 10. For the second-best sector count-wise (point 4), which company received the highest investment? | SST Inc. (Formerly ShotSpotter) | Celltick Technologies	| Manthan Systems |

## Checkpoint 6: Plots

### Gathering data for verification of plot 1 in Tableau 

In [18]:
# total investments (globally) in venture
print("Venture: ", master_frame[(master_frame['funding_round_type']=="venture")]['funding_round_permalink'].count())
# total investments (globally) in seed
print("Seed: ", master_frame[(master_frame['funding_round_type']=="seed")]['funding_round_permalink'].count())
# total investments (globally) in private equity
print("Private Equity: ", master_frame[(master_frame['funding_round_type']=="private_equity")]['funding_round_permalink'].count())

Venture:  47786
Seed:  21024
Private Equity:  1819


### Gathering data for verification of plot 2 in Tableau 

In [19]:
# Displaying country-wise sum of raised_amount_usd in descending order
# We have used the lambda function to convert scientific notation to more readable format
top9.groupby('country_code')['raised_amount_usd'].sum().sort_values(ascending=False).map(lambda x: '{:,.2f}'.format(x))

country_code
USA    420,068,029,342.00
CHN     39,338,918,773.00
GBR     20,072,813,004.00
IND     14,261,508,718.00
CAN      9,482,217,668.00
FRA      7,226,851,352.00
ISR      6,854,350,477.00
DEU      6,306,921,981.00
JPN      3,167,647,127.00
Name: raised_amount_usd, dtype: object

### Gathering data for verification of plot 3 in Tableau 

In [20]:
print("USA")
print(D1.groupby(['main_sector'])['funding_round_permalink'].count().sort_values(ascending=False)[0:3])
print("GBR")
print(D2.groupby(['main_sector'])['funding_round_permalink'].count().sort_values(ascending=False)[0:3])
print("IND")
print(D3.groupby(['main_sector'])['funding_round_permalink'].count().sort_values(ascending=False)[0:3])

USA
main_sector
Others                                     2950
Social, Finance, Analytics, Advertising    2714
Cleantech / Semiconductors                 2350
Name: funding_round_permalink, dtype: int64
GBR
main_sector
Others                                     147
Social, Finance, Analytics, Advertising    133
Cleantech / Semiconductors                 130
Name: funding_round_permalink, dtype: int64
IND
main_sector
Others                                     110
Social, Finance, Analytics, Advertising     60
News, Search and Messaging                  52
Name: funding_round_permalink, dtype: int64


### Exporting master_frame, D1, D2, D3 to excel for lading in Tableau directly (if needed)

In [21]:
# This code will take longer time to execute and must be run only if it is required.
# Using ExcelWriter function of pandas to export dataframes to excel file (filename = InvestmentCaseGroupProject.xlsx)
writer = pd.ExcelWriter('InvestmentCaseGroupProject.xlsx')
# Exporting each dataframe to different worksheet in excel
# Exporting selected field that could be used for plotting
master_frame[['company_permalink','funding_round_permalink','funding_round_type','raised_amount_usd','name','country_code','main_sector']].to_excel(writer,'master_frame')
D1.to_excel(writer,'D1')
D2.to_excel(writer,'D2')
D3.to_excel(writer,'D3')
writer.save()